src/WebUI/WorkLoads/WorkloadDetails.tsx

```diff
@@ -0,0 +1,207 @@
+    constructor(props: IWorkloadDetailsProperties) {
+        super(props, {});
+        this.state = {
+            pods: [],
+            selectedPod: null,
+            showSelectedPod: false
+        };
+        this._podsStore = StoreManager.GetStore<PodsStore>(PodsStore);
+    }
+
+    public render(): JSX.Element {
+        if (this.state.selectedPod && this.state.showSelectedPod) {
+            const parentName = this.props.parentMetaData.name || "";
+            return (<PodsDetails
+                pods={this.state.pods}
+                parentName={parentName}
+            />);
+        }
+
+        return (
+            <div className="workload-details-content">
+                {this._getMainHeading()}
+                {this._getWorkloadDetails()}
+                {this._getAssociatedPods()}
+            </div>
+        );
+    }
+
+    public componentDidMount(): void {
+        const podList = this._podsStore.getState().podsList;
+        let pods: V1Pod[] = (podList && podList.items || []).filter(pod => {
+            return Utils.isOwnerMatched(pod.metadata, this.props.parentMetaData.uid);
+        });
+
+        this.setState({
+            pods: pods,
+            selectedPod: pods[0]
+        });
+    }
+
+    private _getMainHeading(): JSX.Element | null {
+        const metadata = this.props.parentMetaData;
+        if (metadata) {
+            const headerItem: React.ReactNode = (<h2 className="title-heading">{metadata.name}</h2>);
+            return (
+                <div className="content-main-heading">
+                    <ResourceStatus statusProps={this.props.statusProps} customDescription={headerItem} statusSize={StatusSize.l} />
+                </div>
+            );
+        }
+
+        return null;
+    }
+
+    private static _getColumns(): ITableColumn<any>[] {
+        const columns: ITableColumn<any>[] = [
+            {
+                id: "w-image",
+                name: Resources.ImageText,
+                width: new ObservableValue(360),
+                className: "w-key",
+                minWidth: 250,
+                headerClassName: "workload-details-column-header",
+                renderCell: WorkloadDetails._renderImageCell
+            },
+            {
+                id: "w-labels",
+                name: Resources.LabelsText,
+                width: -100,
+                className: "w-key",
+                minWidth: 200,
+                headerClassName: "workload-details-column-header",
+                renderCell: WorkloadDetails._renderLabelsCell
+            }
+        ];
+        return columns;
+    }
+
+    private _getWorkloadDetails(): JSX.Element | null {
+        const metadata = this.props.parentMetaData;
+        if (metadata) {
+            const pipeline = Utils.getPipelineText(metadata.annotations);
+            const tableItems = new ArrayItemProvider<any>([{}, { podTemplate: this.props.podTemplate, parentMetaData: metadata }]);
```

Comment:
if this has more than one image we should show different text
podTemplate

also we should make this as link

Reply:
Modified Image text according to number of containers, And added to do to make it a link when we have image details view